portfolio/src/main/java/com/google/sps/servlets/ShawarmaServlet.java

```diff
@@ -0,0 +1,33 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.ShawarmaPlace;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * Servlet that returns list of shawarma places.
+ */
+@WebServlet("/shawarma")
+public class ShawarmaServlet extends HttpServlet {
+    private final String SHAWARMA_FILE = "shawarma.json";
+    private final Gson gson = new Gson();
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json");
+        try (final Reader reader = Files.newBufferedReader(Paths.get(
+                        Thread.currentThread().getContextClassLoader().getResource(SHAWARMA_FILE).toURI()))) {
+            ShawarmaPlace[] shawarmaPlaces = gson.fromJson(reader, ShawarmaPlace[].class);
+            response.getWriter().println(gson.toJson(shawarmaPlaces));
+        } catch (Exception e) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND);
```

Comment:
There is a reason for converting the file to `ShawarmaPlace` from JSON and them converting it back to JSON when sending the response. We need to know that the JSON file has a valid representation of a list of `ShawarmaPlace`. Instead of catching for `Exception`, you can catch `JsonIOException` and `JsonSyntaxException` first, returning a special error message in that case ("illegal JSON format", or something like that). Then you can catch the generic `Exception` for other cases.